packages/fxa-auth-server/test/local/senders/email.js

```diff
@@ -239,802 +813,171 @@ describe('lib/senders/email:', () => {
-      it(`recovery code settings info link is in email template output for ${type}`, () => {
-        const url = mailer._generateLowRecoveryCodesLinks(message, type).link;
-
-        mailer.mailer.sendMail = stubSendMail(emailConfig => {
-          assert.include(emailConfig.html, url);
-          assert.include(emailConfig.text, url);
-          assert.notInclude(emailConfig.html, 'utm_source=email');
-          assert.notInclude(emailConfig.text, 'utm_source=email');
-        });
-        return mailer[type](message);
-      });
-    }
-
-    if (typesContainLocationData.has(type)) {
-      const defaultLocation = {
-        city: 'Mountain View',
-        country: 'USA',
-        stateCode: 'CA',
-      };
-
-      if (type === 'verifySecondaryEmail') {
-        it(`original user email data is in template for ${type}`, () => {
-          const message = getLocationMessage(defaultLocation);
-          message.primaryEmail = 'user@email.com';
-          mailer.mailer.sendMail = stubSendMail(emailConfig => {
-            assert.include(emailConfig.html, message.primaryEmail);
-            assert.include(emailConfig.html, message.email);
-            assert.include(emailConfig.text, message.primaryEmail);
-            assert.include(emailConfig.text, message.email);
-          });
-          return mailer[type](message);
-        });
-      }
-
-      it(`ip data is in template for ${type}`, () => {
-        const message = getLocationMessage(defaultLocation);
-
-        mailer.mailer.sendMail = stubSendMail(emailConfig => {
-          assert.include(emailConfig.html, message.ip);
-
-          assert.include(emailConfig.text, message.ip);
-        });
-        return mailer[type](message);
-      });
-
-      it(`location is correct with city, country, stateCode for ${type}`, () => {
-        const location = defaultLocation;
-        const message = getLocationMessage(defaultLocation);
-
-        mailer.mailer.sendMail = stubSendMail(emailConfig => {
-          assert.include(
-            emailConfig.html,
-            `${location.city}, ${location.stateCode}, ${location.country}`
-          );
-          assert.include(
-            emailConfig.text,
-            `${location.city}, ${location.stateCode}, ${location.country}`
-          );
-        });
-        return mailer[type](message);
-      });
-
-      it(`location is correct with city, country for ${type}`, () => {
-        const location = Object.assign({}, defaultLocation);
-        delete location.stateCode;
-        const message = getLocationMessage(location);
-
-        mailer.mailer.sendMail = stubSendMail(emailConfig => {
-          assert.include(
-            emailConfig.html,
-            `${location.city}, ${location.country}`
-          );
-          assert.include(
-            emailConfig.text,
-            `${location.city}, ${location.country}`
-          );
-        });
-        return mailer[type](message);
-      });
+  it('verifyEmail handles no service', () => {
```

Comment:
From this point on the tests are all conventional. In another follow-up I plan to extract the tests for `selectEmailService` to a dedicated module as there are so many of them, it should help make things a bit more manageable.